src/object/cli_obj.c

```diff
@@ -4462,109 +4473,479 @@ obj_size_fetch_cb(const struct dc_object *obj, struct obj_auxi_args *obj_auxi)
 	}
 }
 
+#define SMALL_IOV_THRESHOLD 64
+#define FRAGMENT_SIZE       4096
```

Comment:
Merging & copying 4k IOVs looks overkill to me. I suppose lots of applications like fuse buffered I/O could generate 4k IOVs, I'm afraid merging & copying in such case would impact performance unnecessarily.

I think merging & copying excessive small IOVs (<= 64 bytes) should be good enough.

Reply:
So the 4k should be fragment count but not size, right? Here the "FRAGMENT_SIZE" is defined as IOV length but not the number of IOVs, so we'd reduce the "FRAGMENT_SIZE" to 512 or less and enlarge the "FRAGMENT_COUNT" below?

My concern is that 4k is a special size (page size, block size is defined as 4k on many systems), it won't surprise me that an app always read/write in 4k IOVs (through kernel page cache or some userspace cache).

Reply:
I changed FRAGMENT_SIZE to 512 and increase FRAGMENT_COUNT to 256. and for cxi issue, got following replies from Mohamad:

it does not work like that for libfabric on aurora. DAOS team does not provide the libfabric there and it is provided through the slingshot host stack. ANL can update the libfabric there also if needed. so regardless if we pull that into our libfabric packaging or not, it does not matter since our packaging is not used for Slingshot.

Currently, i leave merging tiny IOVs codes as it there.